core/src/consensus/tendermint/mod.rs

```diff
@@ -880,13 +875,12 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
             return false
         }
 
-        let c = match self.client.read().as_ref().and_then(|weak| weak.upgrade()) {
```

Comment:
It seems that it changes the behavior.
Is it okay?

Reply:
Yes. The client should valid in all places.